src/bin/miri.rs

```diff
@@ -227,8 +227,8 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
         } else {
             let return_code = miri::eval_entry(tcx, entry_def_id, entry_type, &config, None)
                 .unwrap_or_else(|| {
-                    //#[cfg(target_os = "linux")]
-                    //miri::native_lib::register_retcode_sv(rustc_driver::EXIT_FAILURE);
+                    #[cfg(trace)]
+                    miri::native_lib::register_retcode_sv(rustc_driver::EXIT_FAILURE);
```

Comment:
At the very least, add a comment explaining why you are doing this here and nowhere else.

Does putting this into `fn exit` (which will exist after a rebase) suffice?

Reply:
It's probably wise to put it there to be safe, but it should also be here above `abort_if_errors()` since that may exit the program without calling this new exit function. I'll add a comment, though

Reply:
Checked, seems to work if we put it there. Ty!